vms/platformvm/state/diff.go

```diff
@@ -209,6 +209,10 @@ func (d *diff) GetCurrentStakerIterator() (StakerIterator, error) {
 	return d.currentStakerDiffs.GetStakerIterator(parentIterator), nil
 }
 
+func (d *diff) GetStakerRewardAttributes(stakerID ids.ID) (*StakerRewardAttributes, error) {
+	return getStakerRewardAttributes(d, stakerID)
```

Comment:
Should this just be calling the parent state `GetStakerRewardAttributes`? It feels strange that we can return staker reward attributes on a tx that only exists on `Diff`

Reply:
In current implementation this is a "derivative" method. The recursion to parent state happens by `GetTx, which is called internally to `getStakerRewardAttributes`.
So current implementation does not really need these methods; it can just use GetTx and do the processing of the tx at client level.
This PR helps moving P-chain to the merkleDB where we don't want txs to be part of the merkle state.

Reply:
This comment was around the implementation of this method on `Diff`. It does not seem correct to return staker reward attributes if the tx only exists on `Diff` and not on the `parentState`. I think this function should be:


```suggestion
	parentState, ok := d.stateVersions.GetState(d.parentID)
	if !ok {
		return nil, fmt.Errorf("%w: %s", ErrMissingParentState, d.parentID)
	}

	return parentState.GetStakerRewardAttributes(stakerID)
```

Reply:
> if the tx only exists on `Diff` and not on the `parentState`


I don't think this is correct. `getStakerRewardAttributes(d, stakerID)` calls `d.GetTx(stakerID)` internally. `d.GetTx` will look for the tx both in the diff and in the parent state.
I added a couple of UTs named `getStakerRewardAttributes works across layers` and `getChainSubnet works across layers` to show this. Let me know if they are convincing.